driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -103,30 +109,35 @@ public InternalConnection get() {
 
     @Override
     public InternalConnection get(final long timeout, final TimeUnit timeUnit) {
+        PooledConnection pooledConnection;
         try {
             if (waitQueueSize.incrementAndGet() > settings.getMaxWaitQueueSize()) {
                 throw createWaitQueueFullException();
             }
             try {
-                connectionPoolListener.waitQueueEntered(new ConnectionPoolWaitQueueEnteredEvent(serverId));
-                PooledConnection pooledConnection = getPooledConnection(timeout, timeUnit);
-                if (!pooledConnection.opened()) {
-                    try {
-                        pooledConnection.open();
-                    } catch (Throwable t) {
-                        pool.release(pooledConnection.wrapped, true);
-                        if (t instanceof MongoException) {
-                            throw (MongoException) t;
-                        } else {
-                            throw new MongoInternalException(t.toString(), t);
-                        }
+                pooledConnection = getPooledConnection(timeout, timeUnit, true);
```

Comment:
I find the boolean parameter hard to understand.  You can't tell what it means without looking at the method implementation.   Consider instead adding

     connectionPoolListener.connectionCheckOutStarted(new ConnectionCheckOutStartedEvent(serverId));

here, and removing the boolean and event-sending in getPooledConnection.  It's only one line and I think it reads cleaner.